mmv1/third_party/terraform/services/networkservices/resource_network_services_lb_edge_extension_test.go

```diff
@@ -0,0 +1,342 @@
+package networkservices_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+	"github.com/hashicorp/terraform-provider-google/google/envvar"
+)
+
+func TestAccNetworkServicesLbEdgeExtension_networkServicesLbEdgeExtensionBasicUpdate(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"project":       envvar.GetTestProjectFromEnv(),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckNetworkServicesLbEdgeExtensionDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkServicesWasmPlugin_artifactRegistryRepositorySetup(context),
+				Check: resource.ComposeTestCheckFunc(
+					// Upload the compiled plugin code to Artifact Registry
+					testAccCheckNetworkServicesWasmPlugin_uploadCompiledCode(
+						t,
+						"google_artifact_registry_repository.test_repository",
+						"my-wasm-plugin",
+						"v1",
+						"test-fixtures/compiled-package/plugin.wasm",
+						"plugin.wasm",
+					),
+					testAccCheckNetworkServicesWasmPlugin_uploadCompiledCode(
+						t,
+						"google_artifact_registry_repository.test_repository",
+						"my-wasm-plugin-2",
+						"v1",
+						"test-fixtures/compiled-package/plugin.wasm",
+						"plugin.wasm",
+					),
+				),
+			},
+			{
+				Config: testAccNetworkServicesLbEdgeExtension_networkServicesLbEdgeExtensionBasicCreate(context),
+			},
+			{
+				ResourceName:            "google_network_services_lb_edge_extension.default",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "location", "name", "port_range", "target", "terraform_labels"},
+			},
+			{
+				Config: testAccNetworkServicesLbEdgeExtension_networkServicesLbEdgeExtensionBasicUpdate(context),
```

Comment:
update steps should confirm that the resource under test is being updated in place.


```suggestion
				Config: testAccNetworkServicesLbEdgeExtension_networkServicesLbEdgeExtensionBasicUpdate(context),
				ConfigPlanChecks: resource.ConfigPlanChecks{
					PreApply: []plancheck.PlanCheck{
						plancheck.ExpectResourceAction("google_pubsub_topic.foo", plancheck.ResourceActionUpdate),
					},
				},
```